duecredit/collector.py

```diff
@@ -219,25 +222,21 @@ def cite(self, entry, **kwargs):
         # TODO: if cite is invoked but no path is provided -- we must figure it out
         # I guess from traceback, otherwise how would we know later to associate it
         # with modules???
+        path = kwargs.get('path', None)
+        if path is None:
+            raise ValueError('path must be provided')
+
         if isinstance(entry, DueCreditEntry):
             # new one -- add it
             self.add(entry)
-            entry_ = entry
+            entry_ = self._entries[entry.get_key()]
         else:
             entry_ = self._entries[entry]
         entry_key = entry_.get_key()
 
-        # TODO: we must allow the same entry be present in multiple citations, so
-        # RF to do so
-        if entry_key not in self.citations:
-            self.citations[entry_key] = Citation(entry_, **kwargs)
-
-        citation = self.citations[entry_key]
-        citation.count += 1
-        # TODO: update path and use here?
-        if not citation.path:
-            citation.path = kwargs.get('path', None)
-
+        citation = Citation(entry_, **kwargs)
+        # update entry count
+        entry_.count += 1
```

Comment:
so you decided to count on the level of entries - why?  if before we could get counts at the level of citations (thus path -- how much specific path was used) and if necessary at the level of their entires; now we would know only some kind of count of entries.  Thus wondering -- why? ;)